File: leetcode_ly/c++/easy/MajorityElement.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int len = nums.size();
        map<int,int> myMap;
        map<int,int>::iterator it;
        for(int i = 0 ; i < len; i++){
            if((it = myMap.find(nums[i])) == myMap.end()){
                myMap.insert(pair<int,int>(nums[i],1));
            }
            else{
                it->second++;
            }
        }
        for(map<int,int>::iterator rt = myMap.begin(); rt != myMap.end(); rt++){
            if(rt->second > len / 2){
                return rt->first;
            }
            else{
                continue;
            }
        }
    }
};

File: leetcode_ly/java/Array/MajorityElement.java
public class Solution {
    public int majorityElement(int[] nums) {
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            if(map.containsKey(nums[i])){
                int num = map.get(nums[i]);
                num++;
                if(num > nums.length / 2) return nums[i];
                map.put(nums[i],num);
            }
            else{
                if(1 > nums.length / 2) return nums[i];
                map.put(nums[i],1);
            }
        }
        return 0;
    }
}